### DISCUSSION TITLE
How to modify parser to not need to escape forward slashes?

### DISCUSSION THREAD

**Discussion by joekim** [inkle/ink#662] [General]
I'm building a narrative around the science of sleep and breathing. I have a lot of references but it's a little annoying to need to escape all the links

```    
Adults need between 7 and 9 hours of sleep per night. # REF: https:\/\/www.sleepfoundation.org\/how-sleep-works\/how-much-sleep-do-we-really-need 
```

How can I modify the parser to not need to escape the forward slash?

Thanks.

**Comment by joethephish** [comment]
Are you sure you need to escape all the slashes? Isn’t it just the double `//` after https, since that gets interpreted as a comment? If that’s the case then you should also only need to escape the second forward slash?

    https:/\/www.sleepfoundation.org/how-sleep-works/how-much-sleep-do-we-really-need

**Reply by joekim** [reply]
Oh, I didn't know that was the issue. I thought all forward slashes had to be escaped based on examples I saw. Escaping the first `//` is a lot more bearable. But is there some path to allow it? For example, no comments once a tag is started? I could make a little pre-processor hack that rewrite `http(s)*://` as `http:\//`

**Reply by joethephish** [reply]
Hmm, comments are already removed in an initial pass before any other parsing is done, so it would be pretty tricky to change that!

**Reply by joekim** [reply]
Ah ok. I think a preprocessor is a good simple solution then. Thanks.
